### PR TITLE
Initial FrontScreen for Choosing between Seller and Customer

### PR DESCRIPTION
**Pull request by dikshamodi**
I've added the frontscreen which should appear when a user opens the spot app. The screen asks for identity and on authentication, we reach to the frontscreen

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      "longitude": null
+    }
+    this.setLocation = this.setLocation.bind(this)
+  }
+
+  setType(type) {
+    this.setState({
+      type: type,
+      latitude: this.state.latitude,
+      longitude: this.state.longitude
```
Comment (harshil93) [inline frontend/src/App.js:21]: I think setState does a merge, can you check the behavior 

this.setState({
      type: type,
      latitude: this.state.latitude,
      longitude: this.state.longitude
})


this.setState({type: type})

are equivalent

File: frontend/src/App.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    });
+  }
+
+  setLocation() {
+    let microapps = window.microapps;
+    let locationJson;
```
Comment (harshil93) [inline frontend/src/App.js:27]: you don't need this variable outside, move it inside the .then. 

Limit the scope of variables and declare them where are they assigned/getting used.

File: frontend/src/App.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        longitude: locationJson.longitude
+      });
+    }).catch(error => {
+      console.error('Error while getting Location: ', error);
+    });
+  }
+
+  render() {
+    return (
+      <>
```
Comment (harshil93) [inline frontend/src/App.js:42]: What does <> mean ?

File: frontend/src/Pages/FrontScreen.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  componentDidMount() {
+    this.authenticate();
+  }
+
+  authenticate() {
+    const microapps = window.microapps;
+    const request = { nonce: 'nonce typically generated server-side' };
+    microapps.getIdentity(request).then(response => {
+      const decoded = JSON.parse(atob(response.split('.')[1]));
```
Comment (harshil93) [inline frontend/src/Pages/FrontScreen.js:24]: Can you add a comment what are you trying to do, what format response is in ?

File: frontend/src/Pages/FrontScreen.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      this.setState(decoded);
```
Comment (harshil93) [inline frontend/src/Pages/FrontScreen.js:25]: this looks wrong to me, there are two setState calls one after another, can you merge.

File: src/main/java/com/hyperlocal/server/MerchantController.java [hunk 6]
```diff
@@ -41,15 +41,15 @@ public MerchantController() {
    * Merchant object
    */
 
-  @PostMapping("/update/merchant/")
+  @PostMapping("/api/update/merchant/")
```
Comment (harshil93) [inline src/main/java/com/hyperlocal/server/MerchantController.java:44]: Are these java files changes intended for this PR ?

File: frontend/src/App.js (hunk 3 above)
Comment (onish-garg) [reply frontend/src/App.js:42]: https://reactjs.org/docs/fragments.html#short-syntax
It is a short-hand for React.Fragment. It's used as a container when there's no one root element.

File: frontend/src/App.js (hunk 1 above)
Comment (onish-garg) [reply frontend/src/App.js:21]: Tip: In ES6, you can just write `this.setState({type})`. If you do not specify the property name, variable's name is picked as the property name

**Pull request merged**